runtime/src/mimalloc/c/alloc-aligned.c

```diff
@@ -0,0 +1,204 @@
+/* ----------------------------------------------------------------------------
```

Comment:
Better add unmodified allocator in one commit and all our changes in the separate one, to make sure we can update the allocator code.